File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
public class Solution {

    public int lengthOfLongestSubstring(String s) {
        Set<Character> set = new HashSet<>();
        int left = 0;
        int right = 0;
        int res = Integer.MIN_VALUE;

        while (right < s.length()) {
            while (set.contains(s.charAt(right))) {
                set.remove(s.charAt(left));
                left++;
            }
            set.add(s.charAt(right));
            right++;
            res = Math.max(res, right - left);
        }

        return res == Integer.MIN_VALUE ? 0 : res;
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-1.kt
class Solution {

    private var map = mutableMapOf(1 to 1, 2 to 2)

    fun climbStairs(n: Int): Int {
        if (n <= 0) return 0
        if (map.contains(n)) return map[n]!!
        var res = climbStairs(n - 1) + climbStairs(n - 2)
        map[n] = res
        return res
    }
}